### ISSUE TITLE
[4.x]: Money field isn't accessible to screen reader users

### ISSUE THREAD

**Issue by gcamacho079** [craftcms/cms#11011]
### Description
The money field does not have programmatically-associated labels, instructions, error text, etc.

### Steps to reproduce

1. Add money field to a field layout
2. Inspect with Chrome DevTools and [view the Accessibility Pane](https://developer.chrome.com/docs/devtools/accessibility/reference/#pane)

### Expected behavior
The Name should match the visual label, and the Description should contain errors, status, instructions, and other messages referenced in the `describedBy` variable (set in the `fieldHtml` function).

For example, the following shows the Accessibility Pane for the title field (with instructions).
<img width="448" alt="Screenshot showing an input name of Headline Required and description text that reads: 'The primary title of this blog post'" src="https://user-images.githubusercontent.com/17525979/165164323-499ac8cc-cec4-4f4d-94e9-7bdd228ecc02.png">

### Actual behavior
The field does not have a name or description text for screen reader users. The following shows the Accessibility Pane for a money field that has both a visible label and instructions:

<img width="646" alt="Screenshot showing money input field with no accessible name or description." src="https://user-images.githubusercontent.com/17525979/165164258-10a62d62-fba6-4d87-8d20-54cd1876be63.png">

### Craft CMS version

4.0.0-beta.4

### Installed plugins and versions

-
